src/common.h

```diff
@@ -161,12 +161,20 @@ typedef Bag Obj;
 **
 **  'ObjFunc' is the type of a function returning an object.
 */
-#pragma GCC diagnostic push
-#ifndef __cplusplus
-#pragma GCC diagnostic ignored "-Wstrict-prototypes"
+#ifdef _WIN32
+typedef FARPROC ObjFunc;
+#else
+/*
+	#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
+	#	include <unistd.h>
+	#	ifdef _POSIX_VERSION
+	#		typedef void *ObjFunc;
+	#	endif
+	#endif
+It's either that or some configure.ac logic - for now, just pretend we have POSIX semantics all the time.
+*/
+typedef void *ObjFunc;
```

Comment:
There is no point in pretending that POSIX semantics are there or not.
Any system were casting a function pointer to a `void *` won't be easily supported by GAP anyway.
```suggestion
typedef void * ObjFunc;
```